core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -802,7 +885,11 @@ object Infer {
         case GenPattern.WildCard => Infer.pure((pat, Nil))
         case GenPattern.Literal(lit) =>
           val tpe = Type.getTypeOf(lit)
-          instSigma(tpe, sigma, reg).as((pat, Nil))
+          val check = sigma match {
```

Comment:
instSigma was only being called once, but here it is clearly on a Tau type because literals are fully known. I inlined it.